src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,174 @@
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
+        currentKeys.push(newKeyCode)
+        this.createKey(newKeyCode, this.attributesArray)
+      }
+    } else {
+      this.clearFormArray(this.attributesArray)
+      this.createKey(newKeyCode, this.attributesArray)
+      this.keyPress = true
+    }
+  }
+
+  private get attributesArray(): UntypedFormArray {
+    return this.actionForm?.get('data')?.get('keys') as UntypedFormArray
+  }
+
+  private addArrValues(items: string[], formArray: UntypedFormArray): void {
+    for (const item of items) {
+      this.createKey(item, formArray)
+    }
+  }
+
+  private createKey(item: string, formArray: UntypedFormArray): void {
+    const control = new UntypedFormControl(item)
+    formArray.push(control)
+  }
+
+  public selectNewActionTrigger(action: Tv2PartAction): void {
+    this.actionForm.patchValue({ actionId: action.id })
+    this.selectedAction = action
+  }
+
+  public onKeyUp(): void {
+    this.keyPress = false
+  }
+
+  public get isUpdateAction(): boolean {
+    return !!this.selectedActionTrigger && !!this.selectedActionTrigger.id
+  }
+
+  public submit(): void {
+    this.submitted = true
+    if (this.actionForm.invalid) {
+      this.actionForm.markAllAsTouched()
+      return
+    }
+    this.submitting = true
+    const actionTriggerValue = this.prepareSendData(this.actionForm.value)
+    if (this.isUpdateAction) {
+      this.updateActionTrigger({ ...actionTriggerValue, id: this.selectedActionTrigger?.id as string })
+    } else {
+      this.createActionTrigger(actionTriggerValue)
+    }
+  }
+
+  private prepareSendData(formData: ActionTriggerUIForm): CreateActionTrigger<KeyboardTriggerData> {
+    const keysArray: string[] = formData.data?.keys
+    const resultData: CreateActionTrigger<KeyboardTriggerData> = {
+      actionId: formData.actionId,
+      data: { keys: keysArray, label: formData.data.label ? formData.data.label : (this.selectedAction as Tv2PartAction).name },
+    }
+    if (this.selectedAction?.argument) {
+      resultData.data.actionArguments = +formData.data.actionArguments
+    }
+    return resultData
+  }
+
+  private createActionTrigger(actionTrigger: CreateActionTrigger<KeyboardTriggerData>): void {
+    this.actionTriggerService.createActionTrigger(actionTrigger).subscribe({
+      next: () => {
+        this.clearFormArray(this.attributesArray)
+        this.actionForm.reset()
+        this.submitting = false
+      },
+    })
+  }
+
+  private updateActionTrigger(actionTrigger: ActionTrigger<KeyboardTriggerData>): void {
+    this.actionTriggerService.updateActionTrigger(actionTrigger).subscribe({
+      next: () => {
+        this.submitting = false
+      },
+    })
```

Comment:
This can be simplified to
```typescript
    this.actionTriggerService
      .updateActionTrigger(actionTrigger)
      .subscribe(() => (this.submitting = false))
```

Same approach can be applied to `createActionTrigger`.

Reply:
Done